File: design-patterns/go/mediator/general/colleague1.go
// colleague1.go

package main

import "fmt"


type Colleague1 struct {
	*Colleague
}

func NewColleague1(mediator *Mediator) (colleague1 *Colleague1) {
	colleague1 = &Colleague1{}
	colleague1.Colleague = NewColleague(mediator)
	return
}

func (colleague1 *Colleague1) ReceiveMessage(msg string) {
	fmt.Printf("Message received in Colleague1: %s\n",  msg)
}

func (colleague1 *Colleague1) SendMessage(colleague IColleague, msg string) {
	colleague1.mediator.SendMessage(colleague, msg)
}

File: design-patterns/ts/mediator/general/colleague1.ts
// colleague1.ts

import Colleague from "./colleague";
import Mediator from "./mediator";

class Colleague1 extends Colleague {
    constructor(mediator: Mediator) {
        super(mediator);
    }

    sendMessage(colleague: Colleague, msg: string): void {
        this.mediator.sendMessage(colleague, msg);
    }

    receiveMessage(msg: string): void {
        console.log("Message received in Colleague1: " + msg);
    }
}

export default Colleague1;